### PR TITLE
Points DB classes

### PR DESCRIPTION
**Pull request by ifnotak**
Adds:
- docker-compose.yml
- Models:
  - UserPoints
  - Action
  - UserAction
  - Trophy
  - UserTrophy
- Tests

---
Best way to test is to:
1. run `docker-compose up --build`
2. go to: http://localhost:8083/swagger-ui.html and execute the **GET** methods on all the classes to retrieve the data added inside *DataPopulation.java*.

### CODE REVIEW

**Comment by ifnotak** [comment]
@1jeanpaul1 requesting review!

File: src/main/java/com/fcgl/madrid/points/model/Action.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package com.fcgl.madrid.points.model;
+
+import javax.persistence.*;
+import java.time.Instant;
+
+@Entity
+@Table(name = "actions")
```
Comment (fcgl) [inline src/main/java/com/fcgl/madrid/points/model/Action.java:7]: I prefer singular singular table names "action" vs "actions" but don't have a strong opinion. Up to you if you want to update the table names

File: src/main/java/com/fcgl/madrid/points/model/Trophy.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
+package com.fcgl.madrid.points.model;
+
+import javax.persistence.*;
+import java.time.Instant;
+
+@Entity
+@Table(name = "trophies")
+public class Trophy {
```
Comment (fcgl) [inline src/main/java/com/fcgl/madrid/points/model/Trophy.java:8]: missing a parameter to keep track of how many times a user has to complete a certain action. `action_count` in the database design dochttps://docs.google.com/document/d/1fXGAdi6qCu7uIhB6h7vughDXoPy8z5MNVcaSXsQeS04/edit

File: src/main/java/com/fcgl/madrid/points/repository/UserTrophyRepository.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.fcgl.madrid.points.repository;
+
+import com.fcgl.madrid.points.model.UserTrophy;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserTrophyRepository extends JpaRepository<UserTrophy, Long> {
+
+}
```
Comment (fcgl) [inline src/main/java/com/fcgl/madrid/points/repository/UserTrophyRepository.java:10]: Can you add a `newline` at the end of the file?

**Review by fcgl — COMMENTED** [review]
* Tables are created ✅ 
* Endpoints all working ✅ 
Looks good, just made some small comments

File: README.md [hunk 4]
```diff
@@ -15,12 +15,15 @@
...
 1. Download docker for your operating system
-2. From project root run the following commands:
-    * **Build:** `docker build -t point-service .`
-    * **Run:** `docker run -d=true -p 8083:8083 point-service`
+2. From project root run the following commands to build and run the app:
+    * `docker-compose up --build`
+    
+## Swagger
+
+Swagger can be accessed through http://localhost:8083/swagger-ui.html
```
Comment (fcgl) [inline README.md:23]: Love the swagger support. We have swagger on the recommendation microservice, but should definitely put it  in all of them

**Comment by fcgl** [comment]
Closing this PR in favor of this one: https://github.com/fcgl/point-system/pull/4

Only difference is some naming changes

**Pull request closed without merging**